TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java

```diff
@@ -1,53 +1,104 @@
+import org.firstinspires.ftc.teamcode.robotSubSystems.claw.ClawConstants;
+import org.firstinspires.ftc.teamcode.robotSubSystems.claw.ClawState;
+import org.firstinspires.ftc.teamcode.robotSubSystems.drivetrain.Drivetrain;
+import org.firstinspires.ftc.teamcode.robotSubSystems.elevator.Elevator;
+import org.firstinspires.ftc.teamcode.robotSubSystems.elevator.ElevatorStates;
+import org.firstinspires.ftc.teamcode.robotSubSystems.intake.Intake;
+import org.firstinspires.ftc.teamcode.robotSubSystems.intake.IntakeState;
 
 public class SubSystemManager {
 
-    public static RobotState getStateFromJoystick(final boolean[] buttons) {
-        // TODO: adjust to ftc
-        // return (buttons[HardwareMap.psCircleIndex]) ? RobotState.TRAVEL
-        // : (buttons[HardwareMap.psXIndex]) ? RobotState.INTAKE
-        // : (buttons[HardwareMap.psOptionsIndex] || GlobalData.isPOVPressed) ?
-        // RobotState.CLIMB
-        // : GlobalData.robotState;
-        return RobotState.TRAVEL;
+    public static RobotState state = RobotState.TRAVEL;
+    static ElevatorStates elevatorState = ElevatorStates.INTAKE;
+    static ClawState clawState = ClawState.CLOSE;
+    static ArmState armState = ArmState.BACK;
+    static IntakeState intakeState = IntakeState.STOP;
+
+    private static RobotState getState (Gamepad gamepad){
+        return gamepad.b ? RobotState.TRAVEL : gamepad.a ? RobotState.INTAKE : gamepad.x ? RobotState.DEPLETE : gamepad.left_bumper ? RobotState.CLAWINTAKE : null;
     }
 
-    private static RobotState getStateFromWantedAndCurrent(final boolean[] buttons,
-                                                           final RobotState robotStateFromJoystick) {
-        RobotState state_W = robotStateFromJoystick;
-        switch (GlobalData.robotState) {
-            case INTAKE:
-                break;
 
-            default:
-                break;
+    public static void setState (Gamepad gamepad, Gamepad gamepad2){
+        final RobotState fromDriver = getState(gamepad);
+          if (fromDriver != null){
+              state = fromDriver;
+          }
+          setSubsystemToState(gamepad, gamepad2);
+    }
 
-        }
-        return state_W;
+    public static ElevatorStates getElevatorStateFromSecondDriver (Gamepad gamepad){
+        return elevatorState = gamepad.a ? ElevatorStates.GROUND : gamepad.b ? ElevatorStates.LOW : gamepad.x ? ElevatorStates.MID : gamepad.y ? ElevatorStates.HIGH : null;
     }
-        public static RobotState getRobotState ( final RobotState userWantedState){
-            return getStateFromWantedAndCurrent(GlobalData.driverPressedButtons, userWantedState);
-        }
 
-    private static RobotState prevRobotState = GlobalData.robotState;
 
-    public static void setSubsystemToState(final RobotState robotState) {
-        switch (robotState) {
+    private static void setSubsystemToState(Gamepad gamepad1, Gamepad gamepad2) {
+        elevatorState = getElevatorStateFromSecondDriver(gamepad2);
```

Comment:
Need to do here the same null safety logics we did with the first gamepad

Reply:
done